webapp/src/ts/modules/reports/reports-content.component.html

```diff
@@ -34,107 +34,110 @@
+            </li>
+          </ng-container>
+
+          <li *ngIf="selection.formatted?.errors?.length">
             <label translate>Errors</label>
             <ul>
-              <li ng-repeat="error in selection.formatted.errors">
+              <li *ngFor="let error of selection.formatted.errors">
                 <p>{{error.message}}</p>
               </li>
             </ul>
           </li>
-          <li *ngIf="selection.formatted.sms_message.message">
+          <li *ngIf="selection.formatted.sms_message?.message">
             <label translate>selection.doc.content.raw</label>
             <p>{{selection.formatted.sms_message.message}}</p>
           </li>
-          <li *ngIf="selection.formatted.tasks.length">
+          <li *ngIf="selection.formatted.tasks?.length">
             <label *ngIf="selection.formatted.kujua_message" translate>tasks.0.messages.0.to</label>
             <label *ngIf="!selection.formatted.kujua_message" translate>Automated Reply</label>
             <ul class="task-list">
-              <li ng-repeat="task in selection.formatted.tasks">
+              <li *ngFor="let task of selection.formatted.tasks">
                 <ul>
-                  <li ng-repeat="message in task.messages">
+                  <li *ngFor="let message of task.messages">
                     <p>{{selection.formatted.kujua_message ? message.to : message.message}}</p>
                   </li>
                 </ul>
-                <span class="task-state" ng-bind-html="task | state"></span>
+                <span class="task-state" [innerHTML]="task | state"></span>
               </li>
               <li class="clear"></li>
             </ul>
           </li>
         </ul>
-        <div class="scheduled-tasks" *ngIf="selection.formatted.scheduled_tasks_by_group.length">
+        <div class="scheduled-tasks" *ngIf="selection.formatted.scheduled_tasks_by_group?.length">
           <h2 translate>tasks</h2>
           <ul>
-            <li ng-repeat="group in selection.formatted.scheduled_tasks_by_group">
+            <li *ngFor="let group of selection.formatted.scheduled_tasks_by_group">
               <h3>{{group.name}}</h3>
               <p>
                 <span ng-hide="group.loading">
                   <a class="btn btn-link" ng-click="edit(selection, group)" translate>Edit</a>
                   <a class="btn btn-link" ng-click="mute(selection, group)" ng-show="canMute(group)" translate>Mute</a>
                   <a class="btn btn-link" ng-click="schedule(selection, group)" ng-show="canSchedule(group)" translate>Schedule</a>
                 </span>
-                <span class="inline loader" ng-show="group.loading"></span>
+                <span class="inline loader" *ngIf="group.loading"></span>
               </p>
               <ul class="task-list">
-                <li ng-repeat="task in group.rows | orderBy:'timestamp'">
+                <li *ngFor="let task of group.rows"> <!-- TODO orderby timestamp!!! -->
```

Comment:
Don't forget the orderBy :) or if it's done, then clean the ToDo

Reply:
Oh! I keep forgetting about this one! Will do!.

Reply:
Done!